src/Datadog.Trace.ClrProfiler.Managed/Integrations/Caching/MethodBuilder.cs

```diff
@@ -0,0 +1,251 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using Datadog.Trace.ClrProfiler.Emit;
+using Sigil;
+
+namespace Datadog.Trace.ClrProfiler.Integrations
+{
+    internal class MethodBuilder<TDelegate>
```

Comment:
Builder/Fluent class for resolving method.
Needs a caching mechanism as of yet.